File: fortran/encrypt.f
function to_upper(strIn) result(strOut)
  
  implicit none
  
  character(len=*), intent(in) :: strIn
  character(len=len(strIn)) :: strOut
  integer :: i,j

  do i = 1, len(strIn)
    j = iachar(strIn(i:i))
    if (j>= iachar("a") .and. j<=iachar("z") ) then
      strOut(i:i) = achar(iachar(strIn(i:i))-32)
    else
      strOut(i:i) = strIn(i:i)
    end if
  end do
end function to_upper

program main
  character(100) :: input, to_upper
  integer :: numShift
  print *, "Enter a string"
  READ(*,*) input
  input = to_upper(input)
  print *, "Enter an integer"
  READ(*,*) numShift
  print *, input
end program main

File: erlang/encrypt.erl
-module(encrypt).   
-export([encrypt/2]). 

shift(C, RealShift) ->
	if 		
		C+RealShift =< $Z -> C+RealShift;
		true -> $A + (C+RealShift - $Z) -1
	end.

encrypt(Input, NumShift) -> 
	RealShift = NumShift rem 26,
	io:format("~s~n", [lists:map(fun(C) -> shift(string:to_upper(C), RealShift) end, Input)]).			